sdk/formrecognizer/azure-ai-formrecognizer/src/main/java/com/azure/ai/formrecognizer/ReceiptExtensions.java

```diff
@@ -0,0 +1,137 @@
+        FormField<String> merchantAddress = null;
+        FormField<String> merchantPhoneNumber = null;
+        FormField<Float> subtotal = null;
+        FormField<Float> tax = null;
+        FormField<Float> tip = null;
+        FormField<Float> total = null;
+        FormField<LocalDate> transactionDate = null;
+        FormField<String> transactionTime = null;
+        IterableStream<USReceiptItem> receiptItems = null;
+
+        for (Map.Entry<String, FormField<?>> entry : receipt.getRecognizedForm().getFields().entrySet()) {
+            String key = entry.getKey();
+            FormField<?> fieldValue = entry.getValue();
+            switch (key) {
+                case "ReceiptType":
+                    receiptType = new USReceiptType(key, fieldValue.getConfidence());
+                    break;
+                case "MerchantName":
+                    merchantName = (FormField<String>) fieldValue;
+                    break;
+                case "MerchantAddress":
+                    merchantAddress = (FormField<String>) fieldValue;
+                    break;
+                case "MerchantPhoneNumber":
+                    merchantPhoneNumber = (FormField<String>) fieldValue;
+                    break;
+                case "Subtotal":
+                    subtotal = (FormField<Float>) fieldValue;
+                    break;
+                case "Tax":
+                    tax = (FormField<Float>) fieldValue;
+                    break;
+                case "Tip":
+                    tip = (FormField<Float>) fieldValue;
+                    break;
+                case "Total":
+                    total = (FormField<Float>) fieldValue;
+                    break;
+                case "TransactionDate":
+                    transactionDate = (FormField<LocalDate>) fieldValue;
+                    break;
+                case "TransactionTime":
+                    transactionTime = (FormField<String>) fieldValue;
+                    break;
+                case "Items":
+                    receiptItems = toReceiptItems(fieldValue);
+                    break;
+                default:
+                    break;
+            }
+        }
+        return new USReceipt(receipt.getReceiptLocale(), receipt.getRecognizedForm(), receiptItems, receiptType,
+            merchantName, merchantAddress, merchantPhoneNumber, subtotal, tax, tip, total, transactionDate,
+            transactionTime);
+    }
+
+    /**
+     * Helper method to convert the service level
+     * {@link com.azure.ai.formrecognizer.implementation.models.FieldValue#getValueArray() value items}
+     * to SDK level {@link USReceiptItem receipt items}.
+     *
+     * @param fieldValueItems The strongly typed field values.
+     *
+     * @return A list of {@link USReceiptItem}.
+     */
+    @SuppressWarnings("unchecked")
+    private static IterableStream<USReceiptItem> toReceiptItems(FormField<?> fieldValueItems) {
```

Comment:
For some of these model types, consider using `List` as there's no additional benefit from using `IterableStream` while List provides options to check the size, provide random access with index etc and can easily be converted to stream or iterator.

Reply:
Is it then ok, for some model variables to be returned as List and some as IterableStream gauging somewhat the way it will be used?

Reply:
Updated some to be a List and kept a few as IterableStream depending on their expected usage.